### PR TITLE
Autocomplete Pagination

### LINKED ISSUE #130: Autocomplete support for pagination
I like that the autocomplete type limits the number of options you see by default. 

Im building a terminal ui for a list that I expect to grow but I want type filtering that the autocomplete type provides. I'd also like users to discover choices by navigating with up and down arrows. It would be great if the list paged through options once the user hits the display limit.

### PR DESCRIPTION
**Pull request by DrunkenPoney**
Fixes #130

### CODE REVIEW

**Comment by lumio** [comment]
Awesome PR! One suggestion though. Maybe it would be nice to automatically switch to the next page if you are at the end of one. Vice-versa for the previous page.

File: lib/elements/autocomplete.js [hunk 1]
```diff
@@ -140,13 +149,29 @@ class AutocompletePrompt extends Prompt {
...
   down() {
-    if (this.select >= this.suggestions.length - 1) return this.bell();
+    if (this.select >= this.suggestions[this.page].length - 1) return this.bell();
     this.moveSelect(this.select + 1);
     this.render();
   }
 
   next() {
-    this.moveSelect((this.select + 1) % this.suggestions.length);
+    this.moveSelect((this.select + 1) % this.suggestions[this.page].length);
```
Comment (lumio) [inline lib/elements/autocomplete.js]: As mentioned in the conversation, it would be nice to jump to the next page if we hit the last element.

**Comment by terkelg** [comment]
Thank you so much! Nice one

File: lib/elements/autocomplete.js (hunk 1 above)
Comment (terkelg) [reply lib/elements/autocomplete.js]: Yeah that's a good idea

Comment (DrunkenPoney) [reply lib/elements/autocomplete.js]: Done!

**Comment by terkelg** [comment]
Can you resolve the merge conflicts @DrunkenPoney? 👍

**Comment by DrunkenPoney** [comment]
Sorry didn't want to close the PR

**Comment by terkelg** [comment]
Awesome, thank you – can you look at why the cursor moves down to the second line please?
screen shot 2019-03-02 at 4 37 15 pm

**Comment by softprops** [comment]
I'm testing the latest locally with 

```json
"dependencies": {
    "prompts": "git@github.com:terkelg/prompts.git#e841383",
  }
```

but when I get to the end of the first page of requests with the down arrow I dont get scrolled to the next page.

screen shot 2019-03-03 at 3 23 30 pm

my prompt object looks something like this


```js
            return {
              type: 'autocomplete',
              name: 'template',
              message: 'Select a serverless template',
              choices: templates.map(template => {
                return {
                  title: `${template.name} - ${chalk.gray(
                    template.description
                  )}`,
                  value: template.githubUrl
                };
              }),
              limit: 20,
              suggest: (input, choices) =>
                Promise.resolve(
                  _.orderBy(
                    choices.filter(c => c.title.indexOf(input) !== -1),
                    ['name'],
                    ['asc']
                  )
                )
            };
```

Im using osx with node 11.6.0 if that helps

**Comment by terkelg** [comment]
@DrunkenPoney is the mastermind behind this 😄

**Comment by softprops** [comment]
so I took a closer look at the change set and found that page up/down ( fn + arrow up/down on mac ) is the way to get the the pages. that works. I found an odd ux in scrolling down to the bottom then doing fn + arrow down takes me to the next page but at the last result of the next page. It would feel more natural if the index was reset on each page up down, if you start at the top when paging down or bottom maybe when page up.

File: lib/elements/autocomplete.js (hunk 1 above)
Comment (softprops) [reply lib/elements/autocomplete.js]: I'm testing this on latest master ( e841383 ) and am not experiencing the auto paging when I get to the last result. I posted some comments below

**Comment by terkelg** [comment]
I agree @softprops – Can you make an issue for this?

**Comment by DrunkenPoney** [comment]
@softprops Up and down arrows won't page up or down one but <kbd>tab</kbd> will page down once you reach the end of the page and will return to the first page once the last one is reached. See [here](https://github.com/terkelg/prompts#autocompletemessage-choices-initial-suggest-limit-style)

@terkelg I'm not home actually but when I'll be home I'll update the readme to include the controls for mac users and fix the selected position when you page up/down.

**Comment by terkelg** [comment]
Awesome @DrunkenPoney! We'll release the next version when that's done 👍

**Comment by softprops** [comment]
confirmed tab works exactly as I expected. Thank you for the awesome patch @DrunkenPoney and thank you @terkelg for the awesome plugin!

**Pull request merged**